src/runtime_src/core/edge/user/aie/aied.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ * Author(s): Himanshu Choudhary <hchoudha@xilinx.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sstream>
+#include <iostream>
+
+#include "aied.h"
+#include "core/edge/include/zynq_ioctl.h"
+#include "core/edge/user/shim.h"
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace zynqaie {
+
+Aied::Aied(std::shared_ptr<xrt_core::device> device): mCoreDevice(std::move(device))
+{
+  mPollingThread = std::thread(&Aied::pollAIE, this);
+}
+
+Aied::~Aied()
+{
+
+}
+
+void 
+Aied::pollAIE()
+{
+  ZYNQ::shim *drv = ZYNQ::shim::handleCheck(mCoreDevice->get_device_handle());
+  xclAIECmd cmd;
+
+  while (1) {
+    /* Calling XRT interface to wait for commands */
```

Comment:
Is this a blocking command?  If so great, please do add a comment indicating so.  

If not, then you have a really fast busy loop which you most likely don't wish to have.

Reply:
done